pyro/infer/mcmc/hmc.py

```diff
@@ -0,0 +1,215 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.autograd import Variable
+
+import pyro
+import pyro.distributions as dist
+import pyro.poutine as poutine
+from pyro.distributions.util import torch_ones_like, torch_zeros_like
+from pyro.infer import TracePosterior
+from pyro.util import ng_ones, ng_zeros
+from tests.common import assert_equal
+
+
+def verlet_integrator(z, r, grad_potential, step_size, num_steps):
```

Comment:
Now that this is using dictionaries, I think we should be able to convert the integrator to a `pyro.optim` optimizer and move it to `pyro.optim`

Reply:
Yes, will take a look at the interface and update this.

Reply:
You should be able to crib from `pyro.optim.ClippedAdam`

Reply:
If we do want it to be a PyroOptim object, it should probably return both the newly sampled param values as well as the "loss" (-delta energy). The optimizer would then need an additional `energy` function as argument. That said, the semantics are somewhat different from the other optimizers since we do not in-place modify the parameters based on some objective function. So we could also create a separate class for this in `pyro.optim`. What do you think?